include/deal.II/lac/iterative_inverse.h

```diff
@@ -129,10 +129,9 @@ inline
 void
 IterativeInverse<VectorType>::initialize(const MatrixType &m, const PreconditionerType &p)
 {
-  // dummy variable
-  VectorType *v = 0;
-  matrix = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(m, *v));
-  preconditioner = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(p, *v));
+  VectorType v;
+  matrix = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(m, v));
+  preconditioner = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(p, v));
```

Comment:
Hm, but that's not good: This suggests that we store a reference to a temporary object that goes out of scope right after these two lines. The documentation states that the vector argument isn't actually used except to determine the vector type via template deduction, but that's not clear at the current location.

That's not your fault -- it's just that the `new_pointer_matrix_base` function is broken: it should simply take the vector type as a template argument. I'll open an issue about this.